<a href="https://colab.research.google.com/github/roberthsu2003/__2026_09_21__ajmj__/blob/main/0921/0921_lesson2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
#整數int
print(15)
print(2005)

15
2005


In [4]:
#浮點數float
print(15.0)
print(50.5)

15.0
50.5


In [5]:
#字串str
print("這是第1個字串")
print('這是第2個字串')

這是第1個字串
這是第2個字串


In [13]:
print(5 + 8)
print(90 - 10)
print(4 * 7)
print(7 / 2)
print(7 // 2)
print(7 % 3)
print(3 ** 4)

13
80
28
3.5
3
1
81


In [19]:
#建立變數
m = 5 + 8
n = 90 - 10
result = n - m

print("m=", m)
print('n=', n)
print('result=', result)

m= 13
n= 80
result= 67


In [27]:
#計算使用者輸入的兩個任意數，並顯示兩數相加的總和：
n = float(input('請輸入第一個數值: '))
m = float(input("請輸入第二個數值: "))
total = n + m
print("===========================")
print("第1個數值是", n)
print("第2個數值是", m)
print("2數相加的結果是", total)

請輸入第一個數值: 27.156
請輸入第二個數值: 54.972
第1個數值是 27.156
第2個數值是 54.972
2數相加的結果是 82.128


### 什麼是 `float()`？

`float()` 函數可以將傳入的資料轉換成「浮點數」（帶有小數點的數字）。它最常用於以下幾種情境：

1. **將字串（str）轉換為浮點數**：
   當我們使用 `input()` 讓使用者輸入數值時，Python 預設會將輸入內容當成「字串」（如 `"27.156"`）。若要進行數學運算，必須先用 `float()` 將它轉換為數值。
2. **將整數（int）轉換為浮點數**：
   將整數轉換為浮點數後，數值會自動加上 `.0`。

下面我們來看幾種常見的 `float()` 使用範例：

In [ ]:
# 1. 將整數轉換成浮點數
num_int = 15
num_float1 = float(num_int)
print("整數轉浮點數:", num_float1)  # 輸出 15.0

# 2. 將包含數字的字串轉換成浮點數
str_num = "3.14159"
num_float2 = float(str_num)
print("字串轉浮點數:", num_float2)  # 輸出 3.14159

# 3. 搭配 input() 接收使用者輸入（如您的上一個儲存格）
# 如果使用者輸入 10，float("10") 會成功轉換為 10.0
user_input = "10"
print("輸入轉浮點數:", float(user_input))

# 4. 常見錯誤提示：如果字串內含有非數字字元，轉換會失敗並產生 ValueError
try:
    float("apple")
except ValueError as e:
    print("\n錯誤示範 (無法將非數字字串轉換成 float):", e)

### 實作練習：計算圓柱體體積

圓柱體體積公式：
$$\text{體積} = \pi \times r^2 \times h$$
其中：
* $r$ 為圓柱底面的**半徑**
* $h$ 為圓柱的**高**
* $\pi$ 為圓周率（我們可以使用 Python 內建 `math` 模組中的 `math.pi`，大約是 3.14159）

In [29]:
import math

# 1. 取得使用者輸入並使用 float() 轉換
radius = float(input("請輸入圓柱體的底面半徑 (r): "))
height = float(input("請輸入圓柱體的高 (h): "))

# 2. 計算圓柱體體積
# math.pi 代表圓周率，radius ** 2 代表半徑的平方
volume = math.pi * (radius ** 2) * height

# 3. 顯示計算結果
print("===========================")
print(f"圓柱體半徑: {radius}")
print(f"圓柱體高度: {height}")
print(f"計算出來的圓柱體體積為: {volume:.2f}")  # :.2f 代表四捨五入保留到小數點後兩位

請輸入圓柱體的底面半徑 (r): 10
請輸入圓柱體的高 (h): 100
圓柱體半徑: 10.0
圓柱體高度: 100.0
計算出來的圓柱體體積為: 31415.93
